components/sbm-core/src/main/java/org/springframework/sbm/build/migration/actions/AddGenericRepositoryAction.java

```diff
@@ -43,6 +43,14 @@ public abstract class AddGenericRepositoryAction extends AbstractAction {
     public void apply(ProjectContext context) {
         RepositoryDefinition.RepositoryDefinitionBuilder builder = RepositoryDefinition.builder();
 
+        if (url == null) {
```

Comment:
Could you annotate the non-nullable fields with `@NotNull` (instead)? 
This will be validated with validation API at creation time.